src/zmq.cpp

```diff
@@ -846,25 +851,10 @@ static int zmq_poller_poll (zmq_pollitem_t *items_, int nitems_, long timeout_)
 }
 #endif // ZMQ_HAVE_POLLER
 
-int zmq_poll (zmq_pollitem_t *items_, int nitems_, long timeout_)
```

Comment:
This is a new function, why are you changing all the existing other implementations? Leave them as they are

Reply:
Most of `zmq_poll` and `zmq_ppoll` are identical, the only big difference is the `pselect` call which needs an extra `sigmask` parameter. Factoring out the identical parts prevents code duplication. I can revert if you prefer, of course, but I had imagined it would create an unnecessary maintenance burden.

Reply:
avoiding to break the stable api is more important

Reply:
Reverted `zmq_poll`.